src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,71 @@
+package codeu.controller;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import codeu.model.store.basic.UserStore;
+
+public class AdminServletTest {
+
+  private AdminServlet adminServlet;
+  private HttpServletRequest mockRequest;
+  private HttpSession mockSession;
+  private HttpServletResponse mockResponse;
+  private RequestDispatcher mockRequestDispatcher;
+
+  @Before
+  public void setup() {
+    adminServlet = new AdminServlet();
+
+    mockRequest = Mockito.mock(HttpServletRequest.class);
+    mockSession = Mockito.mock(HttpSession.class);
+    Mockito.when(mockRequest.getSession()).thenReturn(mockSession);
+    
+    mockResponse = Mockito.mock(HttpServletResponse.class);
+    mockRequestDispatcher = Mockito.mock(RequestDispatcher.class);
+    Mockito.when(mockRequest.getRequestDispatcher("/WEB-INF/view/admin.jsp"))
+        .thenReturn(mockRequestDispatcher);
+  }
+
+  // Tests accessing the admin page with an unregistered user
+  @Test
+  public void testDoGet_UnregisteredUserReturnsDeniedAccess() throws IOException, ServletException {
+  	Mockito.when(mockSession.getAttribute("user")).thenReturn(null);
+  		
+  	adminServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.verify(mockRequest).setAttribute("unregistered_user", 
+    	"You must login and be an admin to view this page.");
+    Mockito.verify(mockRequestDispatcher).forward(mockRequest, mockResponse);
+  }
+
+  // Tests accesing the admin page with a user who is not an admin. They should not be allowed access
+  @Test
+  public void testDoGet_NonAdminReturnsDeniedAccess() throws IOException, ServletException {
+    Mockito.when(mockSession.getAttribute("user")).thenReturn("test_username");
+  	
+  	adminServlet.doGet(mockRequest, mockResponse);
```

Comment:
Weird indentation

Reply:
Is there a reason why the indentation shows up different here than in my editor? I'm using Sublime and in the editor it looks like I am following the 2-space indentation rule. I attached a picture of what it looks like in the editor. How can I remedy this?
<img width="732" alt="screen shot 2018-05-24 at 9 15 03 pm" src="https://user-images.githubusercontent.com/6145106/40521057-9d5d3b02-5f97-11e8-8bc1-369178d51588.png">

Reply:
Oops nevermind, found a solution online.